File: services/forms/app/models/dpt4.py
from sqlalchemy import Column, Integer, String, DateTime, Boolean, Text, Numeric
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.sql import func
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

Base = declarative_base()

class DPT4(Base):
    __tablename__ = "tbl_dpt4"

    id = Column(Integer, primary_key=True, index=True)
    ref_user = Column(Integer, nullable=True)
    company_id = Column(Integer, nullable=True)
    cin = Column(String(255), nullable=True)
    company_name = Column(String(500), nullable=True)
    company_address = Column(Text, nullable=True)
    company_email = Column(String(255), nullable=True)

    # Form-specific fields will be added based on the actual form structure
    # This is a template that needs to be customized for each form

    # Audit fields
    created_by = Column(Integer, nullable=False)
    created_on = Column(DateTime, default=func.now())
    updated_by = Column(Integer, nullable=True)
    updated_on = Column(DateTime, onupdate=func.now())
    is_active = Column(Boolean, default=True)

# Pydantic models
class DPT4Base(BaseModel):
    ref_user: Optional[int] = None
    company_id: Optional[int] = None
    cin: Optional[str] = None
    company_name: Optional[str] = None
    company_address: Optional[str] = None
    company_email: Optional[str] = None

    # Form-specific fields will be added based on the actual form structure
    # This is a template that needs to be customized for each form

    class Config:
        from_attributes = True

class DPT4Create(DPT4Base):
    pass

class DPT4Update(DPT4Base):
    pass

class DPT4Response(DPT4Base):
    id: int
    created_by: int
    created_on: datetime
    updated_by: Optional[int] = None
    updated_on: Optional[datetime] = None
    is_active: bool = True

    class Config:
        from_attributes = True

class DPT4View(DPT4Response):
    ref_user_name: Optional[str] = None

File: legacy/ComplyCrafter_Data/DPT4.cs
using AlphaUtil.Core.Models;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ComplyCrafter_Data
{
    [TableName("tbl_dpt4", "vw_dpt4")]
    public class DPT4 : BaseModel
    {
        [Column("ref_user")]
        public int? RefUser { get; set; }

        [Column("company_id")]
        public int? CompanyId { get; set; }

        [Column("cin")]
        public string? Cin { get; set; }

        [Column("company_name")]
        public string? CompanyName { get; set; }

        [Column("company_address")]
        public string? CompanyAddress { get; set; }

        [Column("company_email")]
        public string? CompanyEmail { get; set; }

        [Column("company_type")]
        public string? CompanyType { get; set; }

        [Column("is_gov")]
        public string? IsGov { get; set; }

        [Column("amt")]
        public string? Amt { get; set; }

        [Column("number_of_depositors")]
        public string? NumberOfDepositors { get; set; }

        [Column("deposits_due_but_not_paid")]
        public string? DepositsDueButNotPaid { get; set; }

        [Column("interest_due_there_on_but_not_paid")]
        public string? InterestDueThereOnButNotPaid { get; set; }

        [Column("deposits_due_but_not_claimed")]
        public string? DepositsDueButNotClaimed { get; set; }

        [Column("interest_due_there_on_but_not_claimed")]
        public string? InterestDueThereOnButNotClaimed { get; set; }

        [Column("deposits_not_yet_due_for_repayment")]
        public string? DepositsNotYetDueForRepayment { get; set; }

        [Column("deposits_due_for_repayment_three_months")]
        public string? DepositsDueForRepaymentThreeMonths { get; set; }

        [Column("arrangement_for_repayment_deposits_due")]
        public string? ArrangementForRepaymentDepositsDue { get; set; }

        [Column("resolution_no")]
        public string? ResolutionNo { get; set; }

        [Column("resolution_date")]
        public DateTime? ResolutionDate { get; set; }
    }

    [TableName("tbl_dpt4", "vw_dpt4")]
    public class DPT4View : DPT4
    {
        [Column("ref_user_name")]
        public string? RefUserName { get; set; }
    }

}